### ISSUE TITLE
Running Web Server Causes Slow Closing Of Options Dialog

### ISSUE THREAD

**Issue by PaulWalkerUK** [albar965/littlenavmap#1017]
I've noticed when I'm running the web server, the Options dialog takes about 5 seconds to close. If it was only when I pressed OK, this might be understandable as the changes get saved etc, but this happens both when I OK but also Cancel, which doesn't make much sense since nothing should be changing. If I stop the web server, it closes immediately. I'm running the 64 bit version on Windows 10. Everything else is fine. From the log file, there's a gap in the times that suggests it's doing something with IP addresses:

```
[2023-04-16 1:56:00.266 DEBUG]: Clicked "Cancel"
[2023-04-16 1:56:00.266 DEBUG]: virtual void OptionsDialog::reject()
[2023-04-16 1:56:00.266 DEBUG]: void WebController::stopServer()
[2023-04-16 1:56:00.266 DEBUG]: void WebMapController::deInit()
[2023-04-16 1:56:00.267 DEBUG]: virtual MapPaintWidget::~MapPaintWidget() delete paintLayer
[2023-04-16 1:56:00.267 DEBUG]: virtual atools::util::FileSystemWatcher::~FileSystemWatcher()
[2023-04-16 1:56:00.268 DEBUG]: virtual MapPaintWidget::~MapPaintWidget() delete screenIndex
[2023-04-16 1:56:00.268 DEBUG]: virtual MapPaintWidget::~MapPaintWidget() delete aircraftTrack
[2023-04-16 1:56:00.268 DEBUG]: virtual MapPaintWidget::~MapPaintWidget() delete aircraftTrackLogbook
[2023-04-16 1:56:00.268 DEBUG]: virtual MapPaintWidget::~MapPaintWidget() delete apronGeometryCache
[2023-04-16 1:56:00.268 DEBUG]: virtual MapPaintWidget::~MapPaintWidget() delete mapQuery
[2023-04-16 1:56:00.271 DEBUG]: HttpListener: closed
[2023-04-16 1:56:00.271 DEBUG]: HttpConnectionHandlerPool (0x137dc800): destroyed
[2023-04-16 1:56:00.271 DEBUG]: HttpListener: closed
[2023-04-16 1:56:00.271 DEBUG]: HttpListener: destroyed
[2023-04-16 1:56:00.271 DEBUG]: static void WebApp::deinit()
[2023-04-16 1:56:00.271 DEBUG]: void WebController::stopServer()
[2023-04-16 1:56:00.271 DEBUG]: void WebController::startServer()
[2023-04-16 1:56:00.272 DEBUG]: static void WebApp::init(QObject*, const QString&, const QString&)
[2023-04-16 1:56:00.273 DEBUG]: TemplateLoader: path=C:/Other Program Files/Little Navmap/LittleNavmap-win64-2.8.10/web, codec=UTF-8
[2023-04-16 1:56:00.273 DEBUG]: TemplateCache: timeout=3600000, size=2000000
[2023-04-16 1:56:00.273 DEBUG]: HttpSessionStore: Sessions expire after 3600000 milliseconds
[2023-04-16 1:56:00.273 DEBUG]: StaticFileController: docroot=C:/Other Program Files/Little Navmap/LittleNavmap-win64-2.8.10/web, encoding=UTF-8, maxAge=60000
[2023-04-16 1:56:00.273 DEBUG]: StaticFileController: cache timeout=60000, size=1000000
[2023-04-16 1:56:00.273 DEBUG]: void WebController::startServer() Docroot "C:/Other Program Files/Little Navmap/LittleNavmap-win64-2.8.10/web" port 8965 SSL false
[2023-04-16 1:56:00.273 DEBUG]: void WebMapController::init()
[2023-04-16 1:56:00.273 DEBUG]: void WebMapController::deInit()
[2023-04-16 1:56:00.323 DEBUG]: atools::util::FileSystemWatcher::FileSystemWatcher(QObject*, bool)
[2023-04-16 1:56:00.323 DEBUG]: void MapLayerSettings::reloadFromFile() ":/littlenavmap/resources/config/maplayers.xml"
[2023-04-16 1:56:00.338 DEBUG]: Using sslKeyFile "" sslCertFile ""
[2023-04-16 1:56:00.339 DEBUG]: HttpListener: Listening on port 8965
[2023-04-16 1:56:00.339 INFO ]: void WebController::startServer() Listening on "0.0.0.0" 8965
[2023-04-16 1:56:00.343 DEBUG]: Found IP "fe80::9021:d46e:f8e0:80a9%ethernet_32774"
[2023-04-16 1:56:04.877 DEBUG]: Found valid IP "192.168.56.1" name "theia"
[2023-04-16 1:56:04.878 DEBUG]: Found IP "fe80::fd3c:9dea:ab64:53da%wireless_32768"
[2023-04-16 1:56:04.879 DEBUG]: Found valid IP "192.168.10.100" name "theia.int.terrakat.uk"
[2023-04-16 1:56:04.879 DEBUG]: Found IP "::1"
[2023-04-16 1:56:04.879 DEBUG]: Found IP "127.0.0.1"
[2023-04-16 1:56:04.879 DEBUG]: void WebController::startServer()
[2023-04-16 1:56:04.879 WARN ]: void WebController::startServer() Web server already running
```

**Comment by albar965** [comment]
I don't see the shutdown delay here, maybe a network issue, but I agree, the web server should not be restarted on cancel.

**Comment by albar965** [comment]
I don't see the shutdown delay here, maybe a network issue, but I agree, the web server should not be restarted on cancel.

**Comment by PaulWalkerUK** [comment]
The delay is between these two events:

```
2023-04-16 1:56:00.343 DEBUG]: Found IP "fe80::9021:d46e:f8e0:80a9%ethernet_32774"
[2023-04-16 1:56:04.877 DEBUG]: Found valid IP "192.168.56.1" name "theia"
```

I think you're right, it's a delay in (re-)starting the web server, but the Options dialog won't go away until this is complete (guess the start is happening in the same thread so blocking the GUI until it completes?). But yes, no need to restart it at all on a Cancel

**Comment by albar965** [comment]
The needless restart is now hopefully fixed with dd829548ec74a94bcc4114c677cbdff7077ce004.
